logic.rb

```diff
@@ -0,0 +1,135 @@
+    6- List all rentals for a given person id.
+    7- Exit"
+end
+
+def display_books(books)
+  books.each do |book|
+    puts "Title: #{book.title}, Author: #{book.author}"
+  end
+end
+
+def display_people(people)
+  people.each do |person|
+    puts "[#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age:#{person.age}"
+  end
+end
+
+def create_student(people)
+  print 'Age: '
+  age = gets.chomp.to_i
+  print 'Name: '
+  name = gets.chomp
+  print 'Has parent permission? [Y/N]: '
+  permission_input = gets.chomp
+  case permission_input
+  when 'Y'
+    permission = true
+  when 'N'
+    permission = false
+  end
+
+  people.push(Student.new(nil, age, name: name, parent_permission: permission))
+  puts 'Person created successfully.'
+end
+
+def create_teacher(people)
+  print 'Age: '
+  age = gets.chomp.to_i
+  print 'Name: '
+  name = gets.chomp
+  print 'specialization: '
+  specialization = gets.chomp
+
+  people.push(Teacher.new(specialization, age, name: name))
+  puts 'Person created successfully.'
+end
+
+def create_book(books)
+  print 'Title: '
+  title = gets.chomp
+  print 'Author: '
+  author = gets.chomp
+
+  books.push(Book.new(title, author))
+  puts 'Book created successfully.'
+end
+
+def create_rental(books, people, rentals)
+  puts 'Select a book  from the following list by number: '
+  books.each_with_index do |book, index|
+    puts "#{index}) Title: #{book.title}, Author: #{book.author}"
+  end
+  book_index = gets.chomp.to_i
+
+  puts 'Select a person from the following list by number (Not ID): '
+  people.each_with_index do |person, index|
+    puts "#{index}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age:#{person.age}"
+  end
+  people_index = gets.chomp.to_i
+
+  print 'Date: '
+  date = gets.chomp
+  rentals.push(Rental.new(date, people[people_index], books[book_index]))
+  puts 'Rental created successfully.'
+end
+
+def display_rental_by_person_id(rentals)
+  print 'Person ID: '
+  person_id = gets.chomp.to_i
+  rentals.each do |rent|
+    puts "Date: #{rent.date}, Book: #{rent.book.title} by #{rent.book.author}." if rent.person.id == person_id
+  end
+end
+
+def student_or_teacher(people)
+  puts 'Do you want to create a student (1) or a teacher (2)? [Input a number]: '
+  student_or_teacher = gets.chomp.to_i
+  case student_or_teacher
+  when 1
+    create_student(people)
+  when 2
+    create_teacher(people)
+  end
+end
+
+def switch_options(option, books, people, rentals)
+  case option
+  when 1
+    display_books(books)
+  when 2
+    display_people(people)
+  when 3
+    student_or_teacher(people)
+  when 4
+    create_book(books)
+  when 5
+    create_rental(books, people, rentals)
+  when 6
+    display_rental_by_person_id(rentals)
+  else
+    puts 'Not an option'
+  end
+end
+
+def switch_case(books, people, rentals)
+  loop do
+    display_list
+    option = gets.chomp.to_i
+    break if option == 7
+
+    switch_options(option, books, people, rentals)
+  end
+end
```

Comment:
- [ ] Good job :clap: On putting some logic in a separate file however the requirement also says that you should put this logic in class(es)

![image](https://user-images.githubusercontent.com/65068771/157129365-33a6ae6f-52d8-42a9-8579-e50375cad61c.png)

Reply:
@andersonlebon Kindly explain more, Do you want to put them inside one class, or more than one. 
because what I understand is that these are just functions that don't need to be inside a class. 
Please if you can give me an example of what you need.

Reply:
hello, @keroloussamy  thanks for reaching out.

Reply:
I can understand that you use only function but the requirement says that you should refactor your code using class by putting some of your logic in classes. [these are your project requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_refactor.md)

Reply:
exemple: 
```rb

```ruby
require "./book.rb"
require "./student.rb"
require "./teacher.rb"
require "./rental.rb"

class App
  def initialize
    @books = []
    @people = []
    @rentals = []
  end
  
  def list_books
    @books.each do |book|
      puts book
    end
  end
  
  def list_people
    @people.each do |person|
      puts person
    end
  end

def create_book
......your logic
end
........
.........
end
```

Reply:
@andersonlebon Okay, thanks for your assistance.